include/hyper.deal/operators/advection/advection_operation.h

```diff
@@ -303,31 +308,61 @@ namespace hyperdeal
             dealii::AlignedVector<VNumber> scratch_data_array;
             scratch_data_array.resize_fast(dealii::Utilities::pow(n_points, dim) * dim_x);
             VNumber *tempp = scratch_data_array.begin();
+            
+            if(do_skew)
```

Comment:
Why not skip the variable `do_skew` and simply check
```suggestion
            if(factor_skew != 0.0)
```
(I would also check for `!=` because in principle one could imagine negative factors even though I'm not sure it makes sense.)